cmd/subnetcmd/addValidator.go

```diff
@@ -165,8 +151,96 @@ func promptDuration(start time.Time) (time.Duration, error) {
 	}
 }
 
+func getMaxValidationTime(network models.Network, nodeID ids.NodeID, startTime time.Time) (time.Duration, error) {
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, constants.RequestTimeout)
+
+	uri := constants.MainnetAPIEndpoint
+	if network == models.Fuji {
+		uri = constants.FujiAPIEndpoint
+	}
+
+	platformCli := platformvm.NewClient(uri)
+	vs, err := platformCli.GetCurrentValidators(ctx, avago_constants.PrimaryNetworkID, nil)
+	cancel()
+	if err != nil {
+		return 0, err
+	}
+	for _, v := range vs {
+		if v.NodeID == nodeID {
+			return time.Unix(int64(v.EndTime), 0).Sub(startTime), nil
+		}
+	}
+	return 0, errors.New("nodeID not found in validator set: " + nodeID.String())
+}
+
+func getTimeParameters(network models.Network, nodeID ids.NodeID) (time.Time, time.Duration, error) {
+	var (
+		start time.Time
+		err   error
+	)
+
+	const (
+		defaultStartOption    = "Start in five minutes"
+		defaultDurationOption = "Until primary network validator expires"
+		custom                = "Custom"
+	)
+
+	if startTimeStr == "" {
+		ux.Logger.PrintToUser("When should your validator start validating?\n" +
+			"If you validator is not ready by this time, subnet downtime can occur.")
+
+		startTimeOptions := []string{defaultStartOption, custom}
+		startTimeOption, err := app.Prompt.CaptureList("Start time", startTimeOptions)
+		if err != nil {
+			return time.Time{}, 0, err
+		}
+
+		switch startTimeOption {
+		case defaultStartOption:
+			start = startTimeDefault
```

Comment:
I think you are not setting this to now() + 1 minute stuff

Reply:
I mean, this start shall be set just before creating the tx, to guarrante the minute

Reply:
great suggestion